// react-use/lib/useGeolocation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("react");
const useGeolocation = () => {
    const [state, setState] = react_1.useState({
        accuracy: null,
        altitude: null,
        altitudeAccuracy: null,
        heading: null,
        latitude: null,
        longitude: null,
        speed: null,
        timestamp: Date.now(),
    });
    let mounted = true;
    let watchId;
    const onEvent = (event) => {
        if (mounted) {
            setState({
                accuracy: event.coords.accuracy,
                altitude: event.coords.altitude,
                altitudeAccuracy: event.coords.altitudeAccuracy,
                heading: event.coords.heading,
                latitude: event.coords.latitude,
                longitude: event.coords.longitude,
                speed: event.coords.speed,
                timestamp: event.timestamp,
            });
        }
    };
    react_1.useEffect(() => {
        navigator.geolocation.getCurrentPosition(onEvent);
        watchId = navigator.geolocation.watchPosition(onEvent);
        return () => {
            mounted = false;
            navigator.geolocation.clearWatch(watchId);
        };
    }, [0]);
    return state;
};
exports.default = useGeolocation;
