// react-use/lib/useMotion.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("react");
const util_1 = require("./util");
const defaultState = {
    acceleration: {
        x: null,
        y: null,
        z: null,
    },
    accelerationIncludingGravity: {
        x: null,
        y: null,
        z: null,
    },
    rotationRate: {
        alpha: null,
        beta: null,
        gamma: null,
    },
    interval: 16,
};
const useMotion = (initialState = defaultState) => {
    const [state, setState] = react_1.useState(initialState);
    react_1.useEffect(() => {
        const handler = (event) => {
            const { acceleration, accelerationIncludingGravity, rotationRate, interval } = event;
            setState({
                acceleration: {
                    x: acceleration.x,
                    y: acceleration.y,
                    z: acceleration.z
                },
                accelerationIncludingGravity: {
                    x: accelerationIncludingGravity.x,
                    y: accelerationIncludingGravity.y,
                    z: accelerationIncludingGravity.z
                },
                rotationRate: {
                    alpha: rotationRate.alpha,
                    beta: rotationRate.beta,
                    gamma: rotationRate.gamma,
                },
                interval
            });
        };
        util_1.on(window, 'devicemotion', handler);
        return () => {
            util_1.off(window, 'devicemotion', handler);
        };
    }, [0]);
    return state;
};
exports.default = useMotion;
